aptos-move/aptos-vm/src/move_vm_ext/write_op_converter.rs

```diff
@@ -40,12 +46,28 @@ macro_rules! convert_impl {
 }
 
 impl<'r> WriteOpConverter<'r> {
-    convert_impl!(convert_resource, get_resource_state_value_metadata);
-
     convert_impl!(convert_module, get_module_state_value_metadata);
 
     convert_impl!(convert_aggregator, get_aggregator_v1_state_value_metadata);
 
+    pub(crate) fn convert_resource(
```

Comment:
I feel you can just pass layout to the macro itself, just make it none for modules and aggregators?

Reply:
Then, I will have to add None to all modules and aggregators before calling convert function, and then I need to remove None again from the output of convert function. It doesn't save any lines of code. Expanding on the macro seems cleaner.

Reply:
I think the current structure is a bit ugly. What if we map `Op<(Bytes, MoveTypeLayout)>` to (Op<Bytes>, MoveTypeLayout)`?